src/parser.ts

```diff
@@ -2412,13 +2427,13 @@ function parseVariableDeclaration(options) {
     const id = parsePattern(params, 'var');
 
     // ECMA-262 12.2.1
-    if (scanner.strict && scanner.isRestrictedWord(id.name)) {
+    if (state.strict && scanner.isRestrictedWord(id.name)) {
```

Comment:
In a follow up, might be a good idea to turn `state.strict && scanner.isRestrictedWord(token)` into its own function, it's used in enough places.